### PR TITLE
Fix: Refresh Skin

### PR DESCRIPTION
**Pull request by jay-fivem**
might need a bit more testing, worked for me

### CODE REVIEW

File: client/main.lua [hunk 1]
```diff
@@ -2099,39 +2099,20 @@ end)
...
-
+    	local ped = PlayerPedId()
+	local model = GetEntityModel(ped)
+        RequestModel(model)
+	Wait(1000) -- Safety Waits
+    	SetPlayerModel(PlayerId(), model)
+    	SetModelAsNoLongerNeeded(model) 
+	Wait(1000) -- Safety Waits
+   	TriggerServerEvent("qb-clothes:loadPlayerSkin") -- LOADING PLAYER'S CLOTHES
+    	TriggerServerEvent("qb-clothing:loadPlayerSkin") -- LOADING PLAYER'S CLOTHES - Event 2
```
Comment (thekierandyer) [inline client/main.lua:2110]: Could you remove this please? Unknown Event

File: client/main.lua [hunk 2]
```diff
@@ -2099,39 +2099,20 @@ end)
...
-
-
-    local playerPed = PlayerPedId()
-    local maxhealth = GetEntityMaxHealth(playerPed)
-    local health = GetEntityHealth(playerPed)
-
-    reloadSkin(health)
-
+    	local ped = PlayerPedId()
+	local model = GetEntityModel(ped)
```
Comment (thekierandyer) [inline client/main.lua:2103]: Might just be me being a defensive programmer, but could you add some logic in the event that GetEntityModel(ped) returns null please?

Eg set model to mp_m_freemode_01/mp_f_freemode_01

**Review by thekierandyer — CHANGES_REQUESTED** [review]
Other than the 2 changes i've listed, LGTM.

File: client/main.lua [hunk 3]
```diff
@@ -2099,39 +2099,20 @@ end)
...
-    SetPlayerModel(PlayerId(), model)
-    SetModelAsNoLongerNeeded(model)
-    Citizen.Wait(1000) -- Safety Delay
-    
-    TriggerServerEvent("qb-clothes:loadPlayerSkin") -- LOADING PLAYER'S CLOTHES
-    TriggerServerEvent("qb-clothing:loadPlayerSkin") -- LOADING PLAYER'S CLOTHES - Event 2
-
-    SetPedMaxHealth(PlayerId(), maxhealth)
-    Citizen.Wait(1000) -- Safety Delay
-    SetEntityHealth(PlayerPedId(), health)
```
Comment (thekierandyer) [inline client/main.lua:2135]: I noticed when testing that if I /refreshskin'd whilst injured it set my health to 100 again. Could you re-add the SetEntityHealth stuff please (in the event is fine)

**Review by ItsANoBrainer — CHANGES_REQUESTED** [review]
This was my solution that properly saves Armor and Health. 
@jay-fivem Lets change it to this and then we can approve.
```lua
function RefreshSkin()
    local ped = PlayerPedId()
    local id = PlayerId()
    local maxhealth = GetPedMaxHealth(ped)
    local health = GetEntityHealth(ped)

    local maxarmour = GetPlayerMaxArmour(id)
    local armour = GetPedArmour(ped)

    local model = GetHashKey("mp_m_freemode_01")

    if PlayerData.charinfo.gender == 1 then -- Gender is ONE for FEMALE
        model = GetHashKey("mp_f_freemode_01") -- Female Model
    end 

    for _, v in pairs(GetGamePool('CObject')) do
        if IsEntityAttachedToEntity(ped, v) then
                SetEntityAsMissionEntity(v, true, true)
                DeleteObject(v)
                DeleteEntity(v)
        end
    end

    LoadPlayerModel(model)
    SetPlayerModel(id, model)
    SetModelAsNoLongerNeeded(model)

    Wait(1000) -- Safety Delay
    
    TriggerServerEvent("qb-clothes:loadPlayerSkin") -- Current Event
    TriggerServerEvent("qb-clothing:loadPlayerSkin") -- Legacy event?

    Wait(1000) -- Safety Delay

    ped = PlayerPedId()
    SetPedMaxHealth(ped, maxhealth)
    SetEntityHealth(ped, health)

    SetPlayerMaxArmour(id, maxarmour)
    SetPedArmour(ped, armour)
end
```

**Pull request closed without merging**